dialer.go

```diff
@@ -52,26 +57,67 @@ func NewDialer(ctx context.Context, opts ...DialerOption) (*Dialer, error) {
 	for _, opt := range opts {
 		opt(&cfg)
 	}
-
 	client, err := sqladmin.NewService(context.Background(), cfg.sqladminOpts...)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create sqladmin client: %v", err)
 	}
+
+	dialCfg := dialCfg{
+		ipType:       cloudsql.IP_TYPE_PUBLIC,
+		tcpKeepAlive: defaultTCPKeepAlive,
+	}
+	for _, opt := range cfg.dialOpts {
+		opt(&dialCfg)
+	}
+
 	d := &Dialer{
-		instances: make(map[string]*cloudsql.Instance),
-		sqladmin:  client,
-		key:       key,
+		instances:      make(map[string]*cloudsql.Instance),
+		sqladmin:       client,
+		key:            key,
+		defaultDialCfg: dialCfg,
 	}
 	return d, nil
 }
 
 // Dial creates an authorized connection to a Cloud SQL instance specified by it's instance connection name.
```

Comment:
Could we add a short bit on what an instance connection name is, e.g., the instance connection name takes the form my-project:us-central1:my-db-instance.